### PR TITLE
Fix nasty holo owner code

### PR DESCRIPTION
**Pull request by thegrb93**

### CODE REVIEW

**Review by Divran — APPROVED** [review]
not tested but code looks good

**Comment by Divran** [comment]
I noticed that the `blocked` table in the `wire_holograms_block_client` console command was indexed by userID and is now indexed by the player. while you're modifying this anyway, could you update it to be indexed by either UID or SteamID64 so that it persists through disconnects? otherwise if someone is being annoying with holograms and you block them, they can simply rejoin to bypass the block

**Comment by thegrb93** [comment]
Should make it a DTVar too instead of a NWVar

**Comment by thegrb93** [comment]
It's working now

**Comment by thegrb93** [comment]
Probably would be better to store the steamid on the entity in client once its player is valid. The only downside being that newly joined player clients won't have the steamid of holograms if the owner is invalid.

File: lua/entities/gmod_wire_hologram.lua [hunk 1]
```diff
@@ -4,15 +4,31 @@ ENT.PrintName = "Wire Hologram"
...
+		if ply:IsValid() then
+			self.steamid = ply:SteamID()
+		end
+	elseif SERVER and not ply:IsValid() then
+		local newply = player.GetBySteamID(self.steamid)
+		if newply then
+			self:SetPlayer(newply)
+			ply = newply
+		end
+	end
```
Comment (thegrb93) [inline lua/entities/gmod_wire_hologram.lua:24]: This needs to be replaced with a join hook that finds holograms with invalid owner and sets them if the id matches.

**Pull request merged**